cluster/v1beta1/types_managedclusterset.go

```diff
@@ -38,6 +38,41 @@ type ManagedClusterSet struct {
 
 // ManagedClusterSetSpec describes the attributes of the ManagedClusterSet
 type ManagedClusterSetSpec struct {
+	// ClusterSelector represents a selector of ManagedClusters
+	// +optional
+	ClusterSelector ManagedClusterSelector `json:"clusterSelector,omitempty"`
+}
+
+// ManagedClusterSelector represents a selector of ManagedClusters
+type ManagedClusterSelector struct {
+	// SelectorType could only be ExclusiveLabel now, will support more SelectorType later
+	// "ExclusiveLabel" means to use a particular cluster label. It guarantees that clustersets with same label key are exclusive
+	// +kubebuilder:validation:Enum=ExclusiveLabel
+	// +kubebuilder:default:="ExclusiveLabel"
+	// +optional
+	SelectorType SelectorType `json:"selectorType"`
+
+	// ExclusiveLabel defines one label which clusterset could use to select target managedClusters. In this way, we will:
+	// 1. Guarantee clustersets with same label key are exclusive
+	// 2. Enable additional permission check when cluster joining/leaving a clusterset.
+	// Currentlly, When a user want to add a label "cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>" for managedCluster,
+	// he/she should have "create" permission for subresource "managedclusters/label" with name "cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>".
+	// +optional
+	ExclusiveLabel *ManagedClusterLabel `json:"exclusiveLabel,omitempty"`
+}
+
+type SelectorType string
+
+const (
+	ExclusiveLabel SelectorType = "ExclusiveLabel"
+)
```

Comment:
Maybe, we could use another `word` to replace the `ExclusiveLabel`.
There are three options: `ClusterLabel`, `SingleLabel`, `ExclusiveLabel`.

`ClusterLabel` or `SingleLabel` will be easily understand for user, but could not contain the Exclusive meaning.
`ExclusiveLabel` can not be easily understood by user. and it's only exclusive when the Label Key is same for different sets.

I am not sure which one is better, or anyone has other options?